src/python/pants/jvm/target_types.py

```diff
@@ -259,3 +260,42 @@ class DeployJarTarget(Target):
         "The JAR will contain class files for both first-party code and "
         "third-party dependencies, all in a common directory structure."
     )
+
+
+# -----------------------------------------------------------------------------------------------
+# `jvm_war` targets
+# -----------------------------------------------------------------------------------------------
+
+
+class JvmWarDependenciesField(Dependencies):
+    pass
+
+
+class JvmWarDescriptorAddressField(SingleSourceField):
+    alias = "descriptor"
+    default = "web.xml"
+    help = "Path to a file containing the descriptor (i.e., web.xml) for this WAR file. Defaults to `web.xml`."
+
+
+class JvmWarContentField(SpecialCasedDependencies):
+    alias = "content"
+    help = (
+        "A list of addresses to `resources` and `files` targets with content to place in the "
+        "document root of this WAR file."
+    )
+
+
+class JvmWarTarget(Target):
+    alias = "jvm_war"
```

Comment:
@Eric-Arellano / @chrisjrn / @stuhood : Bike shed on the name `jvm_war`. I like that it has the technology name in it. But this is inconsistent with `deploy_jar` not having a `jvm_` prefix.

Reply:
I've thought that `deploy_jar` is the outlier in our naming. Did we consider other names for that target? 

--

Here, maybe `war_binary`? Fits with `go_binary`, `pex_binary`, `pyoxidizer_binary`.

Reply:
It's not really a binary though since it is not executable in and of itself. ("war" actually stands for "web application archive.")

Reply:
Interestingly, one set of Bazel rules for producing war's just calls the target type `java_war`. https://github.com/bmuschko/rules_java_war

So maybe `jvm_war` is good enough?

Reply:
Yup, `war_archive` is redundant. Also, WAR format is JVM-specific despite the name seemingly being generic to all web development.

Reply:
Ah, given that, should it simply be `war`? I'm fine with `jvm_war` if you think it makes things more clear.

Reply:
My preference is to include the "technology" name as a prefix in target type names. Probably should get @stuhood to weigh in as well.

Reply:
+1 to `jvm_war`.

Sorry, haven't reviewed the rest, but if @chrisjrn is happy with it then I am.

Reply:
@chrisjrn: Your vote needed ^^

Reply:
Fwiw, we decided on `deploy_jar` as a differentiator from other types of `.jar` files -- though we currently only 
support deploy JARs, we might support thin jars later on (cc @Eric-Arellano).

`jvm_war` seems fine for now.